source/application/DbgHelp/DbgHelpStackWalkCallback.cpp

```diff
@@ -207,13 +351,8 @@ void DbgHelpStackWalkCallback::EnumerateParameters()
          * This will be incremented for each found symbol, with it's size.
          */
         DWORD offset = 0;
-        for (DWORD count = 0; (enum_symbols->Next(1, &sym_info, &num_fetched) == S_OK); count++)
+        for (DWORD count = 0; (enum_symbols->Next(1, &sym_info, &num_fetched) == S_OK && num_fetched == 1); count++)

```

Comment:
I wonder if we can wrap all these (this isn't the only file to iterate over DIA elements) in some sort of construct like the [kernel does for linked lists](http://elixir.free-electrons.com/linux/latest/source/include/linux/list.h#L456).